src/manifest.rs

```diff
@@ -197,9 +197,14 @@ impl Manifest {
                 let blocks = select_blocks_by_archive(update.conn, *archive_id)?;
 
                 for (block_id, stored_digest, length, archive_off) in blocks {
-                    buffer.resize(length as _, 0);
-                    archive.read_exact_at(&mut buffer, archive_off)?;
-                    update.blocks.write_all(&buffer)?;
+                    copy_file_range_full(
+                        &mut buffer,
+                        &archive,
+                        archive_off,
+                        &mut update.blocks,
+                        None,
+                        length,
+                    )?;
 
                     let digest = hash(&buffer);
```

Comment:
This does not work as it breaks the integrity check when the kernel bypass is used.